### PR TITLE
cross compile arm7 32bit and arm8 64bit containers

### PR DESCRIPTION
**Pull request by szuecs**
add cross compiled docker images with docker `buildx`

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

### CODE REVIEW

File: packaging/Makefile [hunk 1]
```diff
@@ -32,23 +32,49 @@ clean:
...
 	docker push $(IMAGE)
 
-build.linux: $(SOURCES)
+# https://docs.docker.com/buildx/working-with-buildx/
+# ~/.docker/config.json add: "experimental": "enabled",
+#docker-buildx-arm64: clean build.linux.arm64
+docker-buildx-arm64: build.linux.arm64
+	docker buildx build -t $(ARM64_IMAGE) --platform linux/arm64 -f Dockerfile.arm64 .
+docker-buildx-arm: clean build.linux.arm
+	docker buildx build -t $(ARM_IMAGE) --platform linux/arm/v7 -f Dockerfile.arm .
```
Comment (szuecs) [inline packaging/Makefile:44]: `buildx` requires .docker/config.json with `"experimental": "enabled",
`

File: packaging/Makefile [hunk 2]
```diff
@@ -32,23 +32,49 @@ clean:
...
+	docker push $(ARM_IMAGE)
+
+# https://www.stereolabs.com/docs/docker/building-arm-container-on-x86/
+arm-deps:
+	#sudo apt-get install qemu binfmt-support qemu-user-static
+	docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+	docker run --rm -t arm64v8/ubuntu uname -m
+	docker run --rm -t arm32v7/ubuntu uname -m
+docker-build-arm64: clean build.linux.arm64
+	docker build -t $(ARM64_IMAGE) -f Dockerfile.arm64 .
```
Comment (szuecs) [inline packaging/Makefile:57]: this way we needs qemu...

File: packaging/Makefile [hunk 3]
```diff
@@ -3,8 +3,8 @@
 VERSION            ?= $(shell git rev-parse HEAD)
 REGISTRY           ?= registry-write.opensource.zalan.do/pathfinder
 IMAGE              ?= $(REGISTRY)/skipper:$(VERSION)
-LIGHTSTEP_IMAGE    ?= $(REGISTRY)/skipper-lightstep:$(VERSION)
-OPENTRACING_IMAGE  ?= $(REGISTRY)/skipper-opentracing:$(VERSION)
+ARM64_IMAGE        ?= $(REGISTRY)/skipper-arm64:$(VERSION)
+ARM_IMAGE          ?= $(REGISTRY)/skipper-arm:$(VERSION)
```
Comment (aryszka) [inline packaging/Makefile:7]: I would call it explicitly V7.

File: packaging/Makefile [hunk 4]
```diff
@@ -32,23 +32,49 @@ clean:
...
+docker-buildx-arm: clean build.linux.arm
+	docker buildx build -t $(ARM_IMAGE) --platform linux/arm/v7 -f Dockerfile.arm .
+docker-push-arm64: docker-buildx-arm64
+	docker push $(ARM64_IMAGE)
+docker-push-arm: docker-buildx-arm
+	docker push $(ARM_IMAGE)
+
+# https://www.stereolabs.com/docs/docker/building-arm-container-on-x86/
+arm-deps:
+	#sudo apt-get install qemu binfmt-support qemu-user-static
```
Comment (aryszka) [inline packaging/Makefile:52]: why the comment? Won't it be required?

Comment (aryszka) [reply packaging/Makefile:52]: nvm

File: packaging/Makefile [hunk 5]
```diff
@@ -32,23 +32,49 @@ clean:
...
 	go build -ldflags "-X main.version=$(VERSION) -X main.commit=$(COMMIT_HASH)" -o build/linux/skipper ../cmd/skipper/*.go
 
+build.linux.arm64:
 	GO111MODULE=$(GO111) \
 	GOOS=linux \
 	GOARCH=arm64 \
 	CGO_ENABLED=$(CGO_ENABLED) \
 	go build -ldflags "-X main.version=$(VERSION) -X main.commit=$(COMMIT_HASH)" -o build/linux_arm8/skipper ../cmd/skipper/*.go
 
+build.linux.arm:
```
Comment (aryszka) [inline packaging/Makefile:77]: arm could be explicitly armv7 or arm32 everywhere, like arm64

**Comment by aryszka** [comment]
:+1:

**Comment by aryszka** [comment]
:+1:

**Pull request merged**